src/tests/test-responses.ts

```diff
@@ -267,7 +267,7 @@ export function getRequestHandlers(): RequestHandler[] {
 
   Object.keys(productsResponsesPerUserId).forEach((userId: string) => {
     const body = productsResponsesPerUserId[userId]!;
-    const url = `http://localhost:8000/rcbilling/v1/subscribers/${userId}/products?id=monthly&id=monthly_2`;
+    const url = `http://localhost:8000/rcbilling/v1/subscribers/${userId}/products`;
```

Comment:
Hmm seems like the code still adds the products as query parameters to the url?  https://github.com/RevenueCat/purchases-js/blob/95743588d01dfdf30d595ce835f86efe0ab89a68/src/networking/endpoints.ts#L56

I guess tests may pass since url matching doesn't take into account query parameters I believe... But might be good idea to keep them here so it's clearer that the URL is supposed to have them? (we actually are not including the currency code, which is included in the code though)

Reply:
Well, that was my train of thought but I guess with a different opinion. Since the request matchers ignore the query parameters it seemed miss-leading to keep them. One may think it will only match if the query parameters are present but it is not true. But I can go ahead and update the behavior so that the code takes them into account inside of the match and returns some sort of failure response if they are not there. Curious to see if all the tests will be passing.

Reply:
Right.... I don't feel strongly TBH, just a thought... I'm ok keeping this as is!

Reply:
So, I tried to only return a successful response if those ids were passed, but that makes some tests fail since their requests do not send both ids or none at all.

For the scope of this PR I decided to:
- Remove the query parameters from the request matcher given that they are not ignored and may lead to thinking the request handler is more constrained when it really is not.
- Changed the url passed to the spy `APIGetRequest` so that it includes the full url received which does include the query parameters.

All tests kept working as they were.